scripts/nn/test/grad_check.dml

```diff
@@ -1844,6 +1846,64 @@ softmax = function() {
   }
 }
 
+softmax2d = function() {
+  /*
+   * Gradient check for the 2D softmax layer.
+   */
+  print("Grad checking the 2D softmax layer with L2 loss.")
+
+  # Generate data
+  N = 3  # num examples
+  C = 10  # num classes
+  Hin = 28  # example height
+  Win = 28  # example width
```

Comment:
This is a good test, but I'm going to set `Hin = 5` & `Win = 5` when I merge just so that the test doesn't take quite as long.